### PR TITLE
Add template management commands: login, logout, pull, push

### PR DESCRIPTION
**Pull request by yankunhuang-pku**
Add six commands:
* Pull templates
* Pull Microsoft templates
* Pull sample data
* Push templates
* Login registry
* Logout registry

### CODE REVIEW

File: FHIR-Converter/client/src/core/template-manager/acr-template-manager.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+			cp.execFileSync(this._orasExePath, ['logout', registryName]);
+			return 'Logout succeeded.';
+		} catch (err) {
+			throw new TemplateManagementError(err.stderr.toString());
+		}
+	}
+
+	pullTemplates(imageReference: string, outputFolder: string, force: boolean) {
+ 		try {
+			 const paramList = ['pull', imageReference, outputFolder];
```
Comment (ACMoretxj) [inline FHIR-Converter/client/src/core/template-manager/acr-template-manager.ts:36]: An extra space in the front of this sentence.

Comment (yankunhuang-pku) [reply FHIR-Converter/client/src/core/template-manager/acr-template-manager.ts:36]: Thanks, removed it.

File: FHIR-Converter/client/src/view/common/input-box/input-box.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License in the project root for license information.
+ */
+
+import { globals } from '../../../core/globals';
+import * as vscode from 'vscode';
+
+export async function showInputBox(placeHolder: string, worksapceKey: string) {
+	const inputBoxOption = { placeHolder: placeHolder, ignoreFocusOut: true };
+	let input = globals.settingManager.getWorkspaceState(worksapceKey);
```
Comment (sowu880) [inline FHIR-Converter/client/src/view/common/input-box/input-box.ts:11]: typo

**Review by moria97 — APPROVED** [review]

File: FHIR-Converter/README.md [hunk 3]
```diff
@@ -91,6 +91,20 @@ Currently, the following features for snippet templates editing are supported:
 You can use the command `FHIR Converter: Update template folder (Ctrl + U)` to update the template folder. In addition, you can also set it through the setting panel of VS Code.
 ![update-template-folder](assets/update-template-folder.gif)
 
+### 6. Template management
+If you don’t have templates yet, you can use the command `FHIR Converter: Pull official templates (Ctrl + D)` to pull the official templates publicly released by Microsoft.
+
+If you want to use the private registry, we also provide the following commands for you to use:
+- `FHIR Converter: Login registry (Ctrl + I)`: Login registry, which is used to log in the private registry. First, you need to enter your private registry, and then the login command will run in the terminal using oras tool. We will store any information about your username and password, which will be asked by oras tool interactively.
```
Comment (sowu880) [inline FHIR-Converter/README.md:98]: We will not store any information ... ?

Comment (yankunhuang-pku) [reply FHIR-Converter/README.md]: Fixed it.

**Review by ACMoretxj — APPROVED** [review]

File: FHIR-Converter/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (ACMoretxj) [reply FHIR-Converter/README.md]: Just ignore, it has been fixed.

File: FHIR-Converter/package.nls.json [hunk 4]
```diff
@@ -43,8 +43,8 @@
 	"message.nonEmptyFolderForcePushOrNot": "Non-empty folder! Do you want to force pull? If force, the files with the same name will be overwritten.",
 	"message.force": "Force",
 	"message.cancel": "Cancel",
-	"message.inputYourRegistry": "Input your registry",
-	"message.inputYourImageReference": "Input your image reference",
+	"message.inputYourRegistry": "Input registry name (sample: name.azurecr.io)",
```
Comment (sowu880) [inline FHIR-Converter/package.nls.json:46]: Input registry login server . (or registry URL? )(sample : registryname.azurecr.io)

File: FHIR-Converter/README.md [hunk 5]
```diff
@@ -99,7 +99,7 @@ If you want to use the private registry, we also provide the following commands
 
 - `FHIR Converter: Logout registry (Ctrl + O)`: Logout registry, which is used to log out the private registry. In this command, you only need to enter your private registry.
 
-- `FHIR Converter: Pull templates (Ctrl + L)`: Pull templates, which is used to pull the private registry images, First, you need to enter your image reference, and then select the output folder to store the templates. If the output folder is not empty, a prompt will pop up. If you choose to force overwrite, the files with the same name will be overwritten.
+- `FHIR Converter: Pull templates (Ctrl + L)`: Pull templates, which is used to pull the private registry images. First, you need to enter your image reference, and then select the output folder to store the templates. If the output folder is not empty, a prompt will pop up. If you choose to force overwrite, the files with the same name will be overwritten.
```
Comment (sowu880) [inline FHIR-Converter/README.md:103]: Update readme for pull sample data?

Comment (yankunhuang-pku) [reply FHIR-Converter/README.md:103]: I will make a new PR about readme.

File: FHIR-Converter/package.nls.json (hunk 4 above)
Comment (yankunhuang-pku) [reply FHIR-Converter/package.nls.json:46]: Updated it.

File: FHIR-Converter/client/src/view/common/registry/pull-image.ts [hunk 6]
```diff
@@ -43,7 +43,7 @@ export async function pullImage(imageReference, refineOutput= false) {
 		
 		if (refineOutput) {
 			// Fix the wrong message from engine temporarily
-			output = output.replace('pulled templates to', 'pushed sample data to');
+			output = output.replace('pulled templates to', 'pulled sample data to');
```
Comment (sowu880) [inline FHIR-Converter/client/src/view/common/registry/pull-image.ts:46]: Could we push the process for pullbar and output information outside this function? maybe in pull-official-templates.ts  and pull-sample-data.ts ? Which can simplify function's parameters.

Comment (yankunhuang-pku) [reply FHIR-Converter/client/src/view/common/registry/pull-image.ts:46]: This change only temporarily processes the output of engine and will be removed later, so try to change other places as little as possible. In the next version of engine, I think we can change the message of engine to the more general message like "pulled files to".

**Review by sowu880 — APPROVED** [review]

**Pull request merged**